### PR TITLE
Add binding for Page.pdf()

### LINKED ISSUE #4: Write bindings for Page.pdf()
`Page.pdf()` needs an implementation similar to other several other functions that take options objects, like `Puppeteer.launch()`. This would involve creating a type `Page.pdfOptions` and a `[@bs.obj]` annotated external `Page.makePDFOptions()`. Like the other examples already implemented, we'll wanted to create another function with the same name to shadow the external `Page.makePDFOptions`. It will convert to JS types--in particular `bool` needs to be converted to `Js.boolean`. We can use the helper functions in our `Util` module for that. 

[Docs](https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#pagepdfoptions)

The [Typescript definitions](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/puppeteer/index.d.ts) are a good reference here. They list all the possible values of the string enum `pdfFormat` which we'll probably want to handle using `[@bs.string]`. Searching for "pdf" on that should lead to the relevant definitions.

Also, the Pdf module can be removed as we should just put this all in the Page module.

### PR DESCRIPTION
**Pull request by jihchi**
Fix #4

### CODE REVIEW

File: __tests__/puppeteer_test.re [hunk 1]
```diff
@@ -196,13 +196,13 @@ describe("Page", () => {
       |> Browser.newPage
       |> then_(page => {
            let options = Navigation.makeOptions(~timeout=25000., ());
-           page |> Page.goto("https://google.com", ~options, ());
+           page |> Page.goto("file://" ++ testPagePath, ~options, ());
```
Comment (jihchi) [inline __tests__/puppeteer_test.re]: ## Before
```
Page
    ✓ goto() (325ms)
```
## After
```
Page
    ✓ goto() (68ms)
```

File: src/Page.re [hunk 2]
```diff
@@ -70,6 +94,50 @@ external makeCookie :
...
+               | `A4
+               | `A5
+             ]
+               =?,
+    ~width: Unit.t=?,
+    ~height: Unit.t=?,
+    ~margin: boxModel=?,
+    unit
+  ) =>
+  _ =
```
Comment (zploskey) [inline src/Page.re]: Why not specify the type `pdfOptions` here?

File: src/Page.re [hunk 3]
```diff
@@ -70,6 +94,50 @@ external makeCookie :
   _ =
   "";
 
+[@bs.obj]
+external makeBoxModel :
```
[the remark this answers is not in the document]
Comment (jihchi) [reply src/Page.re]: `makeMarginOptions` ?

[the remark this answers is not in the document]
Comment (jihchi) [reply src/Page.re]: I think just `makeMargin` and `type margin` would be better.

[the remark this answers is not in the document]
Comment (zploskey) [reply src/Page.re]: Agreed.

File: src/Page.re [hunk 4]
```diff
@@ -141,3 +209,6 @@ external emulateMedia :
 external emulateMediaDisable :
   ([@bs.as {json|null|json}] _) => Js.Promise.t(unit) =
   "emulateMedia";
+
+[@bs.send.pipe: t]
+external pdf : pdfOptions => Js.Promise.t(Js_typed_array.ArrayBuffer.t) = "";
```
[the remark this answers is not in the document]
Comment (jihchi) [reply src/Page.re]: Agree. I'll add comment in the code.

File: src/Page.re [hunk 5]
```diff
@@ -52,6 +52,30 @@ type emulateOption = {
   "userAgent": string,
 };
 
+type boxModel = {
```
[the remark this answers is not in the document]
Comment (jihchi) [reply src/Page.re]: Fix in comment cb78ea6

**Pull request merged**